res/controllers/Pioneer-DDJ-400-script.js

```diff
@@ -243,30 +242,44 @@ PioneerDDJ400.beatFxLevelDepthRotate = function(_channel, _control, value) {
     }
 };
 
-PioneerDDJ400.beatFxSelectPreviousEffect = function(_channel, _control, value) {
-    engine.setValue(PioneerDDJ400.focusedFxGroup(), "prev_effect", value);
-};
-
-PioneerDDJ400.beatFxSelectNextEffect = function(_channel, _control, value) {
-    engine.setValue(PioneerDDJ400.focusedFxGroup(), "next_effect", value);
-};
-
 PioneerDDJ400.beatFxLeftPressed = function(_channel, _control, value) {
     if (value === 0) { return; }
 
-    engine.setValue("[EffectRack1_EffectUnit1]", "focused_effect", 1);
+    var focusedEffect = engine.getValue("[EffectRack1_EffectUnit1]", "focused_effect");
+
+    if (focusedEffect <= 1) {
+        focusedEffect = 3;
+    } else {
+        focusedEffect -= 1;
+    }
+
+    engine.setValue("[EffectRack1_EffectUnit1]", "focused_effect", focusedEffect);
```

Comment:
You seem to be competent in javascript. Could you deduplicate this a bit by factoring out the common parts into a function like this:
```javascript
PioneerDDJ400.changeFocusedEffectBy = function (numberOfSteps) {
        var focusedEffect = engine.getValue("[EffectRack1_EffectUnit1]", "focused_effect");
        var numberOfEffectsPerEffectUnit = 3;
        // standard euclidian modulo by use of two plain modulos
        focusedEffect = (((focusedEffect + numberOfSteps) % numberOfEffectsPerEffectUnit) + numberOfEffectsPerEffectUnit ) % numberOfEffectsPerEffectUnit;
        engine.setValue("[EffectRack1_EffectUnit1]", "focused_effect", focusedEffect);
}
```
Then use that function here (and `PioneerDDJ400.changeFocusedEffectBy(1);` for the right button). 
```suggestion
    PioneerDDJ400.changeFocusedEffectBy(-1);
```

Code cleanups like this usually help us developers tremendously to reduce the amount of time needed to review a mapping and also makes it a much more pleasant experience, which results in mappings being reviewed earlier and faster. Thank you.

Reply:
Thanks for the suggestion. I've had to take the one-based indexes of the effects into account, but seems to be cleaner now.

Reply:
Thank you, I didn't notice that.